### PR TITLE
Principal approval form 2021

### PR DESCRIPTION
**Pull request by clareconstantine**
[PLC-480](https://codedotorg.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=PLC&modal=detail&selectedIssue=PLC-480)

The 20-21 version of the Principal Approval! This PR includes the new model, pd factories, and unit tests, as well as updates to the principal approval form so that it matches the [spec](https://docs.google.com/document/d/1hsteb-kb8M_38YdIS3yM4jA8HmeH3kuK0F50bps1Scc/edit?ts=5d8932fa)

Screen Shot 2019-09-26 at 5 57 01 PM
Screen Shot 2019-09-26 at 5 57 23 PM
Screen Shot 2019-09-26 at 5 57 37 PM

### CODE REVIEW

File: dashboard/app/models/pd/application/principal_approval2021_application.rb [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+#  index_pd_applications_on_application_year     (application_year)
+#  index_pd_applications_on_course               (course)
+#  index_pd_applications_on_regional_partner_id  (regional_partner_id)
+#  index_pd_applications_on_status               (status)
+#  index_pd_applications_on_type                 (type)
+#  index_pd_applications_on_user_id              (user_id)
+#
+
+module Pd::Application
+  class PrincipalApproval2021Application < PrincipalApprovalApplicationBase
```
Comment (hacodeorg) [inline dashboard/app/models/pd/application/principal_approval2021_application.rb:38]: Is this a copy of PrincipalApproval1920Application? Are there any notable change in this file?

File: lib/cdo/shared_constants/pd/principal_approval2021_application_constants.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+module Pd
+  module PrincipalApproval2021ApplicationConstants
```
Comment (hacodeorg) [inline lib/cdo/shared_constants/pd/principal_approval2021_application_constants.rb:2]: Same question for this file

File: dashboard/app/models/pd/application/principal_approval2021_application.rb (hunk 1 above)
Comment (clareconstantine) [reply dashboard/app/models/pd/application/principal_approval2021_application.rb:38]: I started by copying last year's file, and then made the changes reflected in the spec (linked in the description). In retrospect, I should have split these two pieces up, but my commits unfortunately aren't chunked in a way that I can split the PR up now.

**Comment by breville** [comment]
Similar observation to https://github.com/code-dot-org/code-dot-org/pull/30928#issuecomment-535448286; it will be great if/when we can get these form changes down to a single file rather than over a dozen!

**Comment by agealy** [comment]
Spacing above (too little) and below (too much) seems off around "if there is a fee for this program..." line. Otherwise LGTM

**Review by agealy — COMMENTED** [review]
Spacing above (too little) and below (too much) seems off around "if there is a fee for this program..." line. Otherwise LGTM

**Review by islemaster — APPROVED** [review]
Amazing!  Thank you for putting this all together!

**Pull request merged**